File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetstaxratepagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsTaxRatePagedResponseHrefModel } from "./codatdatacontractsdatasetstaxratepagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsTaxRatePagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsTaxRatePagedResponseHrefModel)
  current?: CodatDataContractsDatasetsTaxRatePagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsTaxRatePagedResponseHrefModel)
  next?: CodatDataContractsDatasetsTaxRatePagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsTaxRatePagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsTaxRatePagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsTaxRatePagedResponseHrefModel)
  self?: CodatDataContractsDatasetsTaxRatePagedResponseHrefModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_taxratepagedresponselinksmodel.py
import dataclasses
from ..shared import codat_datacontracts_datasets_taxratepagedresponsehrefmodel as shared_codat_datacontracts_datasets_taxratepagedresponsehrefmodel
from dataclasses_json import dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsTaxRatePagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_taxratepagedresponsehrefmodel.CodatDataContractsDatasetsTaxRatePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_datacontracts_datasets_taxratepagedresponsehrefmodel.CodatDataContractsDatasetsTaxRatePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_datacontracts_datasets_taxratepagedresponsehrefmodel.CodatDataContractsDatasetsTaxRatePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_datacontracts_datasets_taxratepagedresponsehrefmodel.CodatDataContractsDatasetsTaxRatePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    